tests/allocate.ispc

```diff
@@ -0,0 +1,19 @@
+// rule: skip on arch=genx32
+// rule: skip on arch=genx64
+
+void noinline check_val(uniform int *uniform alloca_ptr, uniform float RET[], uniform float aFOO[]) {
+    if (alloca_ptr[programIndex] == aFOO[programIndex]) {
+        RET[programIndex] = 0;
+    }
+}
+
+export void f_f(uniform float RET[], uniform float aFOO[]) {
```

Comment:
I suggest testing alloca() in different CFG contexts in the function, not just in entry basic block. I.e. under the if, in the loop (don't do many iterations!).

Reply:
Added new test conditions.